packages/react/src/components/createInlineOverlayComponent.tsx

```diff
@@ -123,23 +167,48 @@ export const createInlineOverlayComponent = <PropType, ElementType>(
         style,
       };
 
+      const appRoot = document.querySelector('ion-app') || document.body;
+
       /**
-       * We only want the inner component
-       * to be mounted if the overlay is open,
-       * so conditionally render the component
-       * based on the isOpen state.
+       * Trigger-based overlays are dependent on the web component being
+       * mounted in order for the trigger element to attach event listeners.
+       *
+       * For framework implementations, we require that `keepContentsMounted` is set to
+       * `true` in order to always mount the web component.
        */
-      return createElement(tagName, newProps, (this.state.isOpen || this.props.keepContentsMounted) ?
-        createElement('div', {
-          id: 'ion-react-wrapper',
-          ref: this.wrapperRef,
-          style: {
-            display: 'flex',
-            flexDirection: 'column',
-            height: '100%'
-          }
-        }, children) :
-        null
+      const keepContentsMounted =
+        this.props.trigger !== undefined || this.props.keepContentsMounted;
+
+      const mountOverlay =
+        this.state.isOpen || this.props.isOpen || this.isDismissing || keepContentsMounted;
```

Comment:
Why do we need to check both `this.state.isOpen` and `this.props.isOpen`?

Reply:
I will need to dig back into this decision. It was deliberate to solve an issue with the overlay not mounting when using the `isOpen` property directly. I will see if I can reproduce the issue that influenced it.